ci/bin/format.dart

```diff
@@ -226,7 +225,7 @@ abstract class FormatChecker {
     );
     final List<WorkerJob> jobs = patches.map<WorkerJob>((String patch) {
       return WorkerJob(
-        <String>['patch', '-p0'],
+        <String>['git', 'apply', '--ignore-space-change'],
```

Comment:
From my understanding, `--ignore-space-change` might ignore real problems. Should we only add `--ignore-space-change` if on `Platform.isWindows`?

Reply:
From my understanding this fixes formatting issues, and `--ignore-space-change` is to skip line ending changes on Windows. If formatting replaced tabs with spaces, would that get ignored here?

> I'm asking because if those problems are present on Windows as well, then making it conditional does not solve the problem, only limits the damage it can done. I'd much rather solve it proper, if possible.

I completely agree 👍

Reply:
I wonder if on Windows only we could enable `--ignore-cr-at-eol` if that's currently causing an issue with the formatter? I assume that on push, we're normalising line endings to unix line endings?

Reply:
The issue this switch is trying to solve comes from `clang-tidy` on Windows. No matter what line endings you feed into it, the output is always CRLF EOF. So [I use `--ignore-cr-at-eol` at patch creation](https://github.com/flutter/engine/pull/36123/files/8a5be011020666c371a8cacb7bba950c0b0be849#diff-b12ae99b59c24a1d8d53205e819e46da40d4d3869a1cdf509a7905450e09ce38R408) in order to skip patching line endings to CRLF for the whole file. 

As for the patch application side, since `--ignore-cr-at-eol` is a `git diff`-only switch, I needed to find some other solution to avoid line ending differences tripping the context matching in `git apply`. If I remove this switch now, git apply says the patch does not apply (obviously, since the context lines in the patch differ in EOL).

I reviewed the [`--ignore-space-change` switch documentation](https://git-scm.com/docs/git-apply#Documentation/git-apply.txt---ignore-space-change) and I still can't see where this can cause a problem. The switch only controls how the application context is being checked, it does not change the patch itself. So if `clang-tidy` generates a properly formatted file, `git diff` will create the proper patch. Thus git apply (I think) cannot "misapply" the patch if the file did not change since running `format`.

Alternatively I can patch the output of `clang-tidy` EOL, but that could become complicated as it's not guaranteed that files on Windows are checked out with CRLF. You can easily check out files with the proper LF EOL, so that needs to be detected before patching the `clang-tidy` output.

Reply:
Ah I see! Makes sense. Not sure why I misread that as `git diff` when it was clearly the patch command, probably mixing it up with the diff bits later. Thanks for clarifying.